src/Unit-2/lesson5/README.md

```diff
@@ -0,0 +1,150 @@
+````
+
+Go ahead! Click [here](http://localhost:8080/databases/Northwind/indexes/Products/ByCategory?&pageSize=128&include=Category) and
+check out this query result.
+
+Yes, what you get when you execute this query a big JSON object, right? The nice thing is that you get all data you need with a single request. The bad thing is you receive a lot more than you need.
+In the example all related category documents will be present in the response, but you will use just one property (Please, consider that
+Category is a small document ...).
+
+What if you could load the documents in the server-side to produce only the information you need?
+
+## Welcome `LoadDocument`
+
+When writing your index definitions you can use the `LoadDocument` function to get information from related documents. 
+
+Let's rewrite the `Products_ByCategory` index using the `LoadDocument` function.
+
+````csharp
+public class Products_ByCategory : AbstractIndexCreationTask<Product, Products_ByCategory.Result>
+{
+    public class Result
+    {
+        public string Category { get; set; }
+        public int Count { get; set; }
+    }
+
+    public Products_ByCategory()
+    {
+        Map = products =>
+            from product in products
+            let categoryName = LoadDocument<Category>(product.Category).Name
+            select new
+            {
+                Category = categoryName,
+                Count = 1
+            };
+
+        Reduce = results =>
+            from result in results
+            group result by result.Category into g
+            select new
+            {
+                Category = g.Key,
+                Count = g.Sum(x => x.Count)
+            };
+    }
+}
+````
+
+What it means? Now we are no longer storing the category `Id`, but the `Name`. So, now we 
+can rewrite our program with no includes.
+
+````csharp
+static void Main(string[] args)
+{
+    using (var session = DocumentStoreHolder.Store.OpenSession())
+    {
+        var query = session
+            .Query<Products_ByCategory.Result, Products_ByCategory>();
+            //.Include(x => x.Category);
+
+        var results = (
+            from result in query
+            select result
+            ).ToList();
+
+        foreach (var result in results)
+        {
+            //var category = session.Load<Category>(result.Category);
+            //Console.WriteLine($"{category.Name} has {result.Count} items.");
+            Console.WriteLine($"{result.Category} has {result.Count} items.");
+        }
+    }
+}
+````
+
+We will still have only one request...
+
+````
+http://localhost:8080/databases/Northwind/indexes/Products/ByCategory?&pageSize=128
+````
+
+... but now we will have a smaller response.
+
+## It's really good, but...
+
+The `LoadDocument` feature is a really awesome one, but it is also something that should
+be used carefully. 
+
+As you know, `LoadDocument` allows you to load another document during index, and use its data in your index and that is great!
```

Comment:
The key reason to want to use LoadDocument in indexes is to allow _searching_ for data from other documents.
Loading data from other documents is better done using Transformers, since they don't have any such costs.
Might be better way to handle doing an non map/reduce index and allow to search for orders by employee name?